src/main.rs

```diff
@@ -43,21 +138,62 @@ async fn receive_data(data: Data<'_>) -> status::Custom<String> //RawHtml<&'stat
                 );
             }
         };
-    let path = match do_migration(data_string.to_string()) {
+
+    let database: tokio::sync::MutexGuard<'_, Connection> = shared_state.lock().await;
```

Comment:
Similar to the previous comment about optimizing lock acquiring. Move this below the `migrate()` call und drop it after `create_and_add_row()`. In this case it may actually a bit more relevant, because `migrate()` while not very long, does take some time to complete.